infrastructure/persistence/db/mongo.go

```diff
@@ -20,7 +21,22 @@ type MongoInstance struct {
 
 func NewMongoDB(config *configs.Config) (*MongoInstance, error) {
 	mongoURI := config.MongoURI + "/" + config.MongoDBName
-	client, err := mongo.NewClient(options.Client().ApplyURI(mongoURI))
+	clientOptions := options.Client().ApplyURI(mongoURI)
+
+	switch config.Phase {
+	case configs.Production:
+		clientOptions.SetHeartbeatInterval(15 * time.Second)
+		clientOptions.SetMaxPoolSize(100)
```

Comment:
This is mere curiosity. What is the basis for determining the pool size?
experience?

Reply:
I don't know this part either.

@mcauto ,@jiwonkim-park 
Do you guys have any idea how to calculate the proper number of db connection pools?

Reply:
> I don't know this part either.
> 
> @mcauto ,@jiwonkim-park Do you guys have any idea how to calculate the proper number of db connection pools?

I recommend measuring and deciding the performance :)

> **"If you think it’s slow, prove it with a benchmark."**
> [the zen of go](https://dave.cheney.net/2020/02/23/the-zen-of-go)
> **“In the face of ambiguity, refuse the temptation to guess.”**
> The Zen of Python, Item 12

https://github.com/codesenberg/bombardier  
https://github.com/rakyll/hey

Reply:
@mcauto Thank you for always sharing good opensources

Reply:
https://link.medium.com/4WqDHttHXrb